File: src/dataset/typescript/raindrops_Exercism-typescript-shybyte/gemini/GeminiFlash_raindrops_v3.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class Raindrops {
    convert(n) {
        let result = '';
        const factors = [
            { factor: 3, sound: 'Pling' },
            { factor: 5, sound: 'Plang' },
            { factor: 7, sound: 'Plong' }
        ];
        for (const { factor, sound } of factors) {
            if (n % factor === 0) {
                result += sound;
            }
        }
        return result || n.toString();
    }
}
exports.default = Raindrops;

File: src/dataset/c/raindrops_exercism-leleah/gemini/GeminiFlash_raindrops_v3.c
#include "raindrops.h"
#include <string.h>
#include <stdio.h>

void convert(char result[], int drops) {
    result[0] = '\0';
    int len = 0;

    if (drops % 3 == 0) {
        strcat(result, "Pling");
        len += 5;
    }
    if (drops % 5 == 0) {
        strcat(result, "Plang");
        len += 5;
    }
    if (drops % 7 == 0) {
        strcat(result, "Plong");
        len += 5;
    }

    if (len == 0) {
        sprintf(result, "%d", drops);
    }
}